.github/workflows/rust.yml

```diff
@@ -16,6 +16,7 @@ concurrency:
 
 env:
   CARGO_TERM_COLOR: always
+  CARGO_HACK_FLAGS: --feature-powerset --features build-nng --exclude-features default --ignore-unknown-features --optional-deps clap
```

Comment:
I like to improve our current Python script and move more logic there to set per-crate flags, then we could add engine-specific flags in the script rather than in the CI.
To work around the highlighted line, we could install `libnng-dev`, so we don't need to build-nng (It works: https://github.com/hashintel/hash/pull/693). To remove the `optiona-deps` flag we need to wait for https://github.com/taiki-e/cargo-hack/issues/153 to be solved.

I'd keep the flags here for now for simplicity, otherwise we would need to check if `${{ matrix.directory }}` contains `"packages/engine"`